seacrowd/sea_datasets/etos/etos.py

```diff
@@ -0,0 +1,206 @@
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "sent_id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "lemmas": datasets.Sequence(datasets.Value("string")),
+                    "upos": datasets.Sequence(datasets.features.ClassLabel(names=self.UPOS_TAGS)),
+                    "xpos": datasets.Sequence(datasets.Value("string")),
+                    "feats": datasets.Sequence(datasets.Value("string")),
+                    "head": datasets.Sequence(datasets.Value("string")),
+                    "deprel": datasets.Sequence(datasets.Value("string")),
+                    "deps": datasets.Sequence(datasets.Value("string")),
+                    "misc": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_seq_label":
+            features = schemas.seq_label_features(self.UPOS_TAGS)
+
+        else:
+            raise NotImplementedError(f"Schema '{self.config.schema}' is not defined.")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """
+        Returns SplitGenerators.
+        """
+
+        train_path = dl_manager.download_and_extract(_URLS)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": train_path,
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """
+        Yields examples as (key, example) tuples.
+        """
+
+        idx = 0
+        with open(filepath, "r", encoding="utf-8") as data_file:
+            tokenlist = list(conllu.parse_incr(data_file))
+            for sent in tokenlist:
+                if "sent_id" in sent.metadata:
+                    sent_id = sent.metadata["sent_id"]
+                else:
+                    sent_id = idx
+
+                tokens = [token["form"] for token in sent]
+
+                if "text" in sent.metadata:
+                    txt = sent.metadata["text"]
+                else:
+                    txt = " ".join(tokens)
+
+                if self.config.schema == "source":
+                    yield idx, {
+                        "sent_id": str(sent_id),
+                        "text": txt,
+                        "tokens": [token["form"] for token in sent],
```

Comment:
```suggestion
                        "tokens": tokens,
```
DRY, as it is already defined before.